File: fsharp/darts/Darts.fs
module Darts

let score (x: double) (y: double): int = 
    match System.Double.Hypot(x, y) with
    | d when d <=  1.0 -> 10
    | d when d <=  5.0 ->  5
    | d when d <= 10.0 ->  1
    | _                ->  0

File: scala/darts/src/main/scala/Darts.scala
object Darts {
  def score(x: Double, y: Double): Int =
    scala.math.hypot(x, y) match
      case z if z <=  1.0 => 10
      case z if z <=  5.0 =>  5
      case z if z <= 10.0 =>  1
      case _              =>  0
}